src/main/java/net/minecraftforge/registries/NewRegistryEvent.java

```diff
@@ -0,0 +1,98 @@
+package net.minecraftforge.registries;
+
+import net.minecraft.core.Registry;
+import net.minecraftforge.eventbus.api.Event;
+import net.minecraftforge.fml.event.IModBusEvent;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.IdentityHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+import java.util.stream.Collectors;
+
+/**
+ * Register new registries when you receive this event through {@link RegistryBuilder} and {@link #create(RegistryBuilder)}.
+ */
+public class NewRegistryEvent extends Event implements IModBusEvent
+{
+    private final List<RegistryData<?>> registries = new ArrayList<>();
+
+    public NewRegistryEvent() {}
```

Comment:
Formatting: braces on newlines.
```suggestion
    public NewRegistryEvent()
    {
    }
```

Reply:
Lex said {} was fine so that's what I'm going with.